### DISCUSSION TITLE
PnP search filter for document library name

### DISCUSSION THREAD

**Discussion by SinghGurd** [microsoft-search/pnp-modern-search#2808] [Q&A] [answered]
Is there a crawled property that can be set as a RefinableString to filter results based on document library name? I want to set it as a filter in PnP Search Filters web part.

**Comment by kasperbolarsen** [comment, marked as answer]
So you are searching for documents using a query like this:  Path:https://[Tenant].sharepoint.com/sites/PnPModernSearch* IsDocument:1 and you would like a filter allowing you to filter using the display name of the document library?
I know that it is possible because I have used it my self, however it was rather cumbersome. It seems like Out of The Box search does not contain the display Name of the library, so I had to add a "DocumentLibraryName" field of type Single line text to the content types on the libraries in question. When the libraries are created ( provisioning engine) I set the default value of the "DocumentLibraryName" field to the display name of the library. And yes, it is a brittle solution as any owner could rename the libraries and it would break the filter.

**Reply by SinghGurd** [reply]
Thank you...I was thinking the same i.e. create a column and record its value for each document. Thank you for sharing, it confirms I am not missing anything,

**Reply by wobba** [reply]
You could look at parentlink and spweburl, and then figure out the library path name with some string manipulation.
